### PR TITLE
[#514] Add MempoolPayload data type

### LINKED ISSUE #514: Need mempool payload type: union of tx, delegation payload and update payload
## Background

The mempool holds ledger objects that can be included into blocks. These same objects are  transmitted into or between mempools via the transaction submission protocol.

In Byron there are three kinds of ledger objects that can be included:
* transactions `ATxPayload`
* delegation certificates `Delegation.APayload`
* update system objects (proposals, votes) `Update.APayload`

(In Shelley this gets simplified down to just transactions, with all special things embedded withing transactions.)

## Proposed Solution

Add a new module with a new type and appropriate instances and helpers that is the sum of these three types.

It requires binary serialisation instances, but there are no backwards compatability concerns since it will be used with a new network protocol.

### PR DESCRIPTION
**Pull request by intricate**
Closes #514.

### CODE REVIEW

File: cardano-ledger/cardano-ledger.cabal [hunk 1]
```diff
@@ -36,6 +36,7 @@ library
                        Cardano.Chain.Epoch.File
                        Cardano.Chain.Epoch.Validation
                        Cardano.Chain.Genesis
+                       Cardano.Chain.MempoolPayload
```
Comment (intricate) [inline cardano-ledger/cardano-ledger.cabal:39]: Could probably do `Cardano.Chain.Mempool.Payload`, if that's nicer.

File: cardano-ledger/src/Cardano/Chain/MempoolPayload.hs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  toCBOR (MempoolTxPayload tp) =
+    encodeListLen 2 <> toCBOR (0 :: Word8) <> toCBOR tp
+  toCBOR (MempoolDlgPayload dp) =
+    encodeListLen 2 <> toCBOR (1 :: Word8) <> toCBOR dp
+  toCBOR (MempoolUpdatePayload up) =
+    encodeListLen 2 <> toCBOR (2 :: Word8) <> toCBOR up
+
+instance FromCBOR MempoolPayload where
+  fromCBOR = do
+    len <- decodeListLenCanonical
```
Comment (ruhatch) [inline cardano-ledger/src/Cardano/Chain/MempoolPayload.hs]: We actually don't use canonical encodings anymore, so you should use `decodeListLen` and `decodeWord8`

**Review by ruhatch — APPROVED** [review]
LGTM!

**Review by nc6 — APPROVED** [review]
LGTM

**Pull request merged**